crawl-ref/source/describe-god.cc

```diff
@@ -548,6 +553,106 @@ static formatted_string _god_wrath_description(god_type which_god)
     return desc;
 }
 
+static formatted_string _beogh_extra_description()
+{
+    formatted_string desc;
+
+    _add_par(desc, "Named Followers:");
+
+    vector<monster*> followers;
+
+    for (monster_iterator mi; mi; ++mi)
+        if (is_orcish_follower(**mi))
+            followers.push_back(*mi);
+    for (auto &entry : companion_list)
+        // if not elsewhere, follower already seen by monster_iterator
+        if (companion_is_elsewhere(entry.second.mons.mons.mid, true))
+            followers.push_back(&entry.second.mons.mons);
+
+    sort(followers.begin(), followers.end(),
+        [] (monster* a, monster* b) { return a->experience > b->experience;});
+
+    bool has_named_followers = false;
+    for (auto mons : followers)
+    {
+        if (!mons->is_named()) continue;
+        has_named_followers = true;
+
+        desc += formatted_string(mons->full_name(DESC_PLAIN).c_str());
+        if (given_gift(mons))
+        {
+            mon_inv_type slot =
+                mons->props.exists(BEOGH_SH_GIFT_KEY) ? MSLOT_SHIELD :
+                mons->props.exists(BEOGH_ARM_GIFT_KEY) ? MSLOT_ARMOUR :
+                mons->props.exists(BEOGH_RANGE_WPN_GIFT_KEY) ? MSLOT_ALT_WEAPON :
+                MSLOT_WEAPON;
+
+            desc.cprintf(" (");
+
+            item_def &gift = mitm[mons->inv[slot]];
+            desc += formatted_string::parse_string(menu_colour_item_name(gift,DESC_PLAIN));
+            desc.cprintf(")");
+        }
+        desc.cprintf("\n");
+    }
+
+    if (!has_named_followers)
+        _add_par(desc, "None");
+
+    return desc;
+}
+
+static string _describe_deck_summary()
+{
+    ostringstream desc;
+    desc << "Decks of power:\n";
+    for (int i = FIRST_PLAYER_DECK; i <= LAST_PLAYER_DECK; i++)
+        desc << " " << deck_status((deck_type) i) << "\n";
+
+    string stack = stack_contents();
+    if (!stack.empty())
+        desc << "\n stacked deck: " << stack << "\n";
+
+    return desc.str();
+}
+
+static formatted_string _god_extra_description(god_type which_god)
+{
+    formatted_string desc;
+
+    switch (which_god)
+    {
+        case GOD_ASHENZARI:
+            if (have_passive(passive_t::bondage_skill_boost))
+            {
+                _add_par(desc, "Curse skill supports:");
```

Comment:
A minor oddity: possibly only WJC's extra screen should show up on the join menu UI, but right now only Ash's is prevented from showing up.